### DISCUSSION TITLE
Fail to search in Homebrew/homebrew-core repository

### DISCUSSION THREAD

**Discussion by ngocphamm** [Homebrew/discussions#4017] [Everyday usage] [answered]
I'm really not sure if this is a GitHub bug, or a setting to exclude search for the repository, but I don't think I can search for anything in the `Homebrew/homebrew-core` repository.

I was trying to figure out why `brew search rust` on my machine doesn't return anything (`Error: No formulae or casks found for "rust".`), and could it because I tried out `HOMEBREW_INSTALL_FROM_API = 1`, and removed the 2 folders `/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core` and `/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask`.

So I ran `brew search rust -v -d`. It prints out the command to search GitHub code.

```
/usr/local/Homebrew/Library/Homebrew/shims/shared/curl --disable --cookie /dev/null --globoff --show-error --user-agent Homebrew/3.6.13-34-g001bace\ \(Macintosh\;\ Intel\ Mac\ OS\ X\ 12.6.1\)\ curl/7.79.1 --header Accept-Language:\ en --retry 3 --location https://api.github.com/search/code\?q=user\%3AHomebrew\+path\%3AFormula\+path\%3ACasks\+path\%3A.\+filename\%3Arust\+extension\%3Arb\&per_page=100 --header Accept:\ application/vnd.github\+json --write-out '
'\%\{http_code\} --header Authorization:\ token\ ****** --dump-header /private/tmp/github_api_headers20221204-93071-tbyq3n
```

So it looks like it does try to search GitHub (and I do have a personal token set for Homebrew). I then play with the search query for a bit, and found out that it doesn't seem to return anything in the repository `Homebrew/homebrew-core`, even from GitHub's own web interface.

https://github.com/Homebrew/homebrew-core/search?q=repo%3AHomebrew%2Fhomebrew-core+extension%3Amd+filename%3AREADME+path%3A%2F&type=Code&ref=advsearch

![CleanShot 2022-12-05 at 07 42 09@2x](https://user-images.githubusercontent.com/395123/205639909-46dad3ef-1c7e-4cc1-8cd3-049e5d445942.png)

Similar query, but targeting `Homebrew/brew` repository, and there's a result.

https://github.com/search?q=repo%3AHomebrew%2Fbrew+extension%3Amd+filename%3AREADME+path%3A%2F&type=Code&ref=advsearch

![CleanShot 2022-12-05 at 07 43 56@2x](https://user-images.githubusercontent.com/395123/205640192-b8d92306-3a3b-4684-a5d9-e4d46d449fcd.png)

**Comment by chenrui333** [comment]
I think it is github issue. (I am using the new github code search)

![image](https://user-images.githubusercontent.com/1580956/205751644-de8ab1a5-04eb-47eb-b623-e485816d57aa.png)

This would be the one that you are probably looking for, https://github.com/search?q=repo%3AHomebrew%2Fhomebrew-core+repo%3AHomebrew%2Fhomebrew-core+path%3A**%2FREADME.md&type=code

`repo:Homebrew/homebrew-core repo:Homebrew/homebrew-core path:**/README.md`

![image](https://user-images.githubusercontent.com/1580956/205752082-6550fe9c-41c5-41a1-ace4-c62b46f4b8b2.png)

**Reply by Bo98** [reply]
Yes, Code Search has been broken on homebrew-core and homebrew-cask for quite a while now. It is unlikely the old search will ever be fixed, so the only solution is to [sign up](https://github.com/features/code-search-code-view/signup) for the new search beta.

For `brew search`, in the future we'll likely stop using GitHub code search in favour of other methods.

**Reply by ngocphamm** [reply]
@Bo98 So you mean once I sign up for new search beta (whenever it gets to me), the search, from my terminal, will be fixed? 🤔 

@chenrui333 Yeah it looks that way to me, but I'm confused at the same time that it seems to work elsewhere. Could that be the size of the 2 repository is too much for it to handle, or there's a setting somewhere in those 2 repositorythat prevents the kind of search.

**Reply by Bo98** [reply]
> @Bo98 So you mean once I sign up for new search beta (whenever it gets to me), the search, from my terminal, will be fixed? 🤔

Not sure if it's actually available via API yet.

> Could that be the size of the 2 repository is too much for it to handle

Probably this. Whatever it is, it's on GitHub's side.

**Comment by ngocphamm** [comment, marked as answer]
Related to this, and is now fixed. https://github.com/Homebrew/brew/issues/14009
